screens/AppNavigator/screens/TransactionsScreen/TransactionsScreen.tsx

```diff
@@ -1,36 +1,149 @@
+      return
+    }
+
+    setStatus('loading')
+    account.getAddress().then(async address => {
+      return await whaleApiClient.address.listTransaction(address, undefined, nextToken)
+    }).then(async addActivities => {
+      setHasNext(addActivities.hasNext)
+      setNextToken(addActivities.nextToken as string|undefined)
+      setAddressActivities([...activities, ...addActivities])
+      setStatus('idle')
+    }).catch((e) => {
+      setError(e)
+      setStatus('error')
+    })
+  }
+
+  useEffect(() => {
+    loadData()
+  }, [])
+
+  const TransactionRow = (row: { item: AddressActivity }): JSX.Element => {
+    let iconName: 'arrow-up' | 'arrow-down'
+    let color: 'green'|'gray'
+    let desc = ''
+    const isPositive = row.item.vin === undefined
+
+    // TODO(@ivan-zynesis): fix when other token transaction can be included
+    const TOKEN_NAME: { [key in number]: string } = {
+      0: 'DFI',
+      1: 'tBTC'
+    }
+
+    const tokenId = TOKEN_NAME[row.item.tokenId as number]
+    let amount = new BigNumber(row.item.value)
+
+    if (isPositive) {
+      color = 'green'
+      // TODO(@ivan-zynesis): Simplified, more complicated token transaction should have different icon and desc
+      iconName = 'arrow-down'
+      desc = 'Received'
+    } else {
+      color = 'gray'
+      iconName = 'arrow-up'
+      desc = 'Sent'
+      amount = amount.negated()
+    }
+
+    const option = {
+      id: row.item.id,
+      desc,
+      iconName,
+      color,
+      amount: amount.toFixed(),
+      block: row.item.block.height,
+      token: tokenId
+    }
+
+    const txIcon = <Ionicons name='arrow-down' size={24} color={color} />
+    const amountText = <NumberText value={option.amount} style={{ color }} />
+
+    return (
+      <View key={row.item.id} style={tailwind('bg-white p-2 border-b border-gray-200 flex-row items-center w-full h-16')}>
+        <View style={tailwind('w-full flex-row flex-initial')}>
+          <View style={tailwind('w-8 justify-center')}>
+            {txIcon}
+          </View>
+          <View style={tailwind('flex-auto flex-row justify-center items-center')}>
+            <View style={tailwind('flex-auto flex-col ml-3 justify-center')}>
+              <Text style={tailwind('font-medium')}>
+                {translate('screens/TransactionsScreen', option.desc)}
+              </Text>
+              <Text style={tailwind('font-medium')}>
+                {translate('screens/TransactionsScreen', 'block')}: {option.block}
+              </Text>
+            </View>
+            <View style={tailwind('flex-row ml-3 items-center')}>
+              {amountText}
+              <View style={tailwind('w-16 ml-3 items-start')}>
+                <Text style={tailwind('flex-shrink font-medium')}>
+                  {option.token}
+                </Text>
+              </View>
+            </View>
+          </View>
+          <View style={tailwind('w-8 flex-grow-0 justify-center flex-end')}>
```

Comment:
`flex-end` is not supported on mobile. 
![image](https://user-images.githubusercontent.com/25013382/122525974-61183380-d04c-11eb-9b61-f29b90c7f14b.png)

Reply:
@thedoublejay any idea what should I used to align the item right in a flex-row?

Reply:
@ivan-zynesis I think in this case, you can make the middle element flex-grow 1. To take the remaining space in between. This way will push the transaction symbol on the left and the caret on the right most. You can delete the `flex-end` I think it should be ok